src/tox/tox_env/python/api.py

```diff
@@ -161,7 +163,22 @@ def _setup_env(self) -> None:
         """setup a virtual python environment"""
         super()._setup_env()
         self.ensure_python_env()
-        self._paths = self.prepend_env_var_path()  # now that the environment exist we can add them to the path
+        self._env_paths = self.prepend_env_var_path()  # now that the environment exist we can add them to the path
+
+    @property
+    def environment_variables(self) -> dict[str, str]:
+        env_vars = super().environment_variables
+        if self._env_paths:
+            values = dict.fromkeys(str(i) for i in self._env_paths)
+            values.update(dict.fromkeys(env_vars["PATH"].split(os.pathsep)))
+            env_vars = {**env_vars, "PATH": os.pathsep.join(values)}
```

Comment:
This is not python specific logic, so it's just wrong to push it here. Needs to be at the base level.